lte/cloud/go/services/had/servicers/servicer.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servicers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"magma/lte/cloud/go/lte"
+	"magma/lte/cloud/go/protos"
+	"magma/lte/cloud/go/serdes"
+	lte_models "magma/lte/cloud/go/services/lte/obsidian/models"
+	"magma/orc8r/cloud/go/orc8r"
+	orc8r_serdes "magma/orc8r/cloud/go/serdes"
+	"magma/orc8r/cloud/go/services/analytics/query_api"
+	"magma/orc8r/cloud/go/services/configurator"
+	"magma/orc8r/cloud/go/services/metricsd/prometheus/restrictor"
+	"magma/orc8r/cloud/go/services/state"
+	"magma/orc8r/cloud/go/services/state/wrappers"
+	"magma/orc8r/lib/go/metrics"
+	orc8rprotos "magma/orc8r/lib/go/protos"
+
+	"github.com/golang/glog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	validSecsSinceStateReported = 180
+)
+
+type HAServicer struct {
+	promClient query_api.PrometheusAPI
+}
+
+// NewHAServicer creates a new service implementing the HaD proto file
+func NewHAServicer(promClient query_api.PrometheusAPI) protos.HaDServer {
+	return &HAServicer{
+		promClient: promClient,
+	}
+}
+
+// GetEnodebOffloadState fetches all primary gateways that the calling gateway
+// is in a gateway pool with. For each of these gateways, it then fetches the
+// offload state for each of the gateway's ENBs.
```

Comment:
Generally, if a method is just implementing an iface (e.g. grpc iface), a doc comment is superfluous. But in this case it might be insightful to a particular reader as a summary of what happens below, so not end of the world